src/client/Microsoft.Identity.Client/Platforms/Android/Broker/AndroidContentProviderBroker.cs

```diff
@@ -0,0 +1,393 @@
+            }
+
+            using (_logger.LogBlockDuration("Waiting for Android broker response. "))
+            {
+                await AndroidBrokerStaticHelper.ReadyForResponse.WaitAsync().ConfigureAwait(false);
+                return AndroidBrokerStaticHelper.InteractiveBrokerTokenResponse;
+            }
+        }
+
+        private async Task AcquireTokenInteractiveViaContentProviderAsync(BrokerRequest brokerRequest)
+        {
+            using (_logger.LogMethodDuration())
+            {
+                _logger.Verbose("Starting Android Broker interactive authentication. ");
+
+                Bundle bundleResult = await PerformContentResolverOperationAsync(ContentResolverOperation.acquireTokenInteractive, null).ConfigureAwait(false);
+
+                var interactiveIntent = CreateInteractiveBrokerIntent(brokerRequest, bundleResult);
+
+                _brokerHelper.LaunchInteractiveActivity(_parentActivity, interactiveIntent);
+            }
+        }
+
+        private Intent CreateInteractiveBrokerIntent(BrokerRequest brokerRequest, Bundle bundleResult)
+        {
+            string packageName = bundleResult.GetString("broker.package.name");
+            string className = bundleResult.GetString("broker.activity.name");
+            string uid = bundleResult.GetString("caller.info.uid");
+
+            Intent brokerIntent = new Intent();
+            brokerIntent.SetPackage(packageName);
+            brokerIntent.SetClassName(
+                    packageName,
+                    className
+            );
+
+            brokerIntent.PutExtras(bundleResult);
+            brokerIntent.PutExtra(BrokerConstants.NegotiatedBPVersionKey, _negotiatedBrokerProtocolKey);
+
+            var interactiveIntent = brokerIntent.PutExtras(CreateInteractiveBrokerBundle(brokerRequest));
+            interactiveIntent.PutExtra(BrokerConstants.CallerInfoUID, Binder.CallingUid);
+
+            return interactiveIntent;
+        }
+
+        private Bundle CreateInteractiveBrokerBundle(BrokerRequest brokerRequest)
+        {
+            _brokerHelper.ValidateBrokerRedirectURI(brokerRequest);
+
+            Bundle bundle = new Bundle();
+            string brokerRequestJson = JsonHelper.SerializeToJson(brokerRequest);
+            bundle.PutString(BrokerConstants.BrokerRequestV2, brokerRequestJson);
+            bundle.PutInt(BrokerConstants.CallerInfoUID, Binder.CallingUid);
+            _logger.InfoPii("GetInteractiveBrokerBundle: " + brokerRequestJson, "Enable PII to see the broker request. ");
+            return bundle;
+        }
+
+        public async Task<MsalTokenResponse> AcquireTokenSilentAsync(
+            AuthenticationRequestParameters authenticationRequestParameters,
+            AcquireTokenSilentParameters acquireTokenSilentParameters)
+        {
+            InitiateBrokerHandShakeAsync();
+
+            BrokerRequest brokerRequest = BrokerRequest.FromSilentParameters(
```

Comment:
this seems the exact same as the AndroidAccoutnManagerBroker class...oh wait...are these two different 'android broker" flavors? i have lots of quesitons for tomorrow. :)

Reply:
Discussed in meeting